tutorials/preprocessing/plot_15_handling_bad_channels.py

```diff
@@ -0,0 +1,209 @@
+# interpolating its signal based on the signals of the good sensors around it.
+#
+#
+# How interpolation works
+# ~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Interpolation of EEG channels in MNE-Python is done using the spherical
+# spline method [1]_, which projects the sensor locations onto a unit sphere
+# and interpolates the signal at the bad sensor locations based on the signals
+# at the good locations. Mathematical details are presented in
+# :ref:`channel_interpolation`. Interpolation of MEG channels uses the field
+# mapping algorithms used in computing the :ref:`forward solution
+# <tut-forward>`.
+#
+#
+# Interpolation in MNE-Python
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Interpolating bad channels in :class:`~mne.io.Raw` objects is done with the
+# :meth:`~mne.io.Raw.interpolate_bads` method, which automatically applies the
+# correct method (spherical splines or field interpolation) to EEG and MEG
+# channels, respectively (there is a corresponding method
+# :meth:`mne.Epochs.interpolate_bads` that works for :class:`~mne.Epochs`
+# objects). To illustrate how it works, we'll start by cropping the raw object
+# to just three seconds for easier plotting:
+
+raw.crop(tmin=0, tmax=3).load_data()
+
+###############################################################################
+# By default, :meth:`~mne.io.Raw.interpolate_bads` will clear out
+# ``raw.info['bads']`` after interpolation, so that the interpolated channels
+# are no longer excluded from subsequent computations. Here, for illustration
+# purposes, we'll prevent that by specifying ``reset_bads=False`` so that when
+# we plot the data before and after interpolation, the affected channels will
+# still plot in red:
+
+eeg_data = raw.copy().pick_types(meg=False, eeg=True, exclude=[])
+eeg_data_interp = eeg_data.copy().interpolate_bads(reset_bads=False)
+
+for data in (eeg_data, eeg_data_interp):
+    data.plot(butterfly=True, color='#00000022', bad_color='r')
+
+###############################################################################
+# Note that we used the ``exclude=[]`` trick in the call to
+# :meth:`~mne.io.Raw.pick_types` to make sure the bad channels were not
+# automatically dropped from the selection. Here is the corresponding example
+# with the interpolated gradiometer channel; since there are more channels
+# we'll use a more transparent gray color this time:
+
+grad_data = raw.copy().pick_types(meg='grad', exclude=[])
+grad_data_interp = grad_data.copy().interpolate_bads(reset_bads=False)
+
+for data in (grad_data, grad_data_interp):
+    data.plot(butterfly=True, color='#00000011', bad_color='r')
+
+###############################################################################
+# References
+# ^^^^^^^^^^
+#
+# .. [1] Perrin, F., Pernier, J., Bertrand, O. and Echallier, JF. (1989).
+#        Spherical splines for scalp potential and current density mapping.
+#        *Electroencephalography Clinical Neurophysiology* 72(2):184-187.
```

Comment:
We could perhaps also add a pointer to autoreject for automating these tasks : http://autoreject.github.io